File: src/test/java/application/history/IntegrationTest.java
package application.history;

import application.history.history.presentation.interfaces.api.GetAllHistoriesController;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.servlet.MockMvc;

import javax.transaction.Transactional;

@SpringBootTest(classes = GetAllHistoriesController.class)
@AutoConfigureMockMvc
@Transactional
public class IntegrationTest {
    @Autowired protected MockMvc mvc;
    @Autowired protected ObjectMapper objectMapper;
}

File: src/test/kotlin/com/example/history/application/history/IntegrationTest.kt
package com.example.history.application.history

import application.history.history.presentation.interfaces.api.GetAllHistoriesController
import javax.transaction.Transactional

@SpringBootTest(classes = [GetAllHistoriesController::class])
@AutoConfigureMockMvc
@Transactional
open class IntegrationTest {
    @Autowired
    protected var mvc: MockMvc? = null

    @Autowired
    protected var objectMapper: ObjectMapper? = null
}
